yt_dlp/extractor/globalplayer.py

```diff
@@ -0,0 +1,272 @@
+from .common import InfoExtractor
+from ..utils import (
+    clean_html,
+    parse_duration,
+    str_or_none,
+    traverse_obj,
+    unified_strdate,
+    unified_timestamp,
+    urlhandle_detect_ext,
+)
+
+
+class GlobalPlayerBaseIE(InfoExtractor):
+
+    def _get_pageProps(self, url, video_id):
+        webpage = self._download_webpage(url, video_id)
+        data = self._search_nextjs_data(webpage, video_id)
+        return traverse_obj(data, ('props', 'pageProps'))
+
+    def _request_ext(self, url, video_id):
+        urlh = self._request_webpage(url, video_id, note='Determining source extension')
+        return urlhandle_detect_ext(urlh)
+
+    def _extract_audio(self, episode, series):
+        return {
+            'vcodec': 'none',
+            **traverse_obj(series, {
+                'series': 'title',
+                'series_id': 'id',
+                'thumbnail': 'imageUrl',
+            }),
+            **traverse_obj(episode, {
+                'id': 'id',
+                'description': ('description', {clean_html}),
+                'duration': ('duration', {parse_duration}),
+                'thumbnail': 'imageUrl',
+                'url': 'streamUrl',
+                'timestamp': (['pubDate', 'startDate'], {unified_timestamp}),
+                # pubDate for podcasts, startDate for radio catchup - that's all we need to have both in one
+                'title': 'title',
+            }, get_all=False)
+        }
+
+
+class GlobalPlayerLiveIE(GlobalPlayerBaseIE):
+    _VALID_URL = r'https?://www\.globalplayer\.com/live/(?P<id>\w+)/\w+/$'
+    _TESTS = [{
+        'url': 'https://www.globalplayer.com/live/smoothchill/uk/',
+        'info_dict': {
+            'id': '2mx1E',
+            'ext': 'aac',
+            'title': str,
+            'thumbnail': 'md5:407a54f3a18e54aa0326a399e68a7d50',
+            'description': 'Music To Chill To',
+            'live_status': 'is_live',
+            'display_id': 'smoothchill-uk',
+        }}, {
```

Comment:
> Are you writing these tests by hand?
yeah, i find that it's easier to put what i'm expecting there myself, and if the tests pass, great, otherwise go fix it

have redone the test formatting with the script, though it also makes some ? decisions imo